integration-tests/testutils/report/parser.go

```diff
@@ -89,19 +91,37 @@ func NewSubunitV2ParserReporter(writer io.Writer) *SubunitV2ParserReporter {
 
 func (fr *SubunitV2ParserReporter) Write(data []byte) (int, error) {
 	var err error
+	sdata := string(data)
 
-	if matches := announceRegexp.FindStringSubmatch(string(data)); len(matches) == 2 {
-		err = fr.statuser.Status(subunit.Event{TestID: matches[1], Status: "exists"})
-	} else if matches := successRegexp.FindStringSubmatch(string(data)); len(matches) == 2 {
-		err = fr.statuser.Status(subunit.Event{TestID: matches[1], Status: "success"})
-	} else if matches := failureRegexp.FindStringSubmatch(string(data)); len(matches) == 2 {
+	if matches := announceRegexp.FindStringSubmatch(sdata); len(matches) == 2 {
+		testID := matches[1]
+		if isTest(testID) {
+			err = fr.statuser.Status(subunit.Event{TestID: matches[1], Status: "exists"})
+		}
+	} else if matches := successRegexp.FindStringSubmatch(sdata); len(matches) == 2 {
+		testID := matches[1]
+		if isTest(testID) {
+			err = fr.statuser.Status(subunit.Event{TestID: matches[1], Status: "success"})
+		}
+	} else if matches := failureRegexp.FindStringSubmatch(sdata); len(matches) == 2 {
 		err = fr.statuser.Status(subunit.Event{TestID: matches[1], Status: "fail"})
-	} else if matches := skipRegexp.FindStringSubmatch(string(data)); len(matches) == 3 {
+	} else if matches := skipRegexp.FindStringSubmatch(sdata); len(matches) == 3 {
+		reason := matches[2]
+		// Do not report anything about the set ups skipped because of another test's reboot.
+		duringReboot, _ := regexp.MatchString(
```

Comment:
As a rule of thumb, errors should never be ignored. When they are, there should be a comment above the line explaining why it's being ignored. If it's hard to justify, don't ignore it. If it's boring to explain, don't ignore it either, as it's easier to check than to explain.

In this case, for example, a broken regular expression will look silently and exactly the same as an expression that did not match. If you're sure the expression always matches, then create a local helper function such as:

```
func matchString(pattern string, s string) (matched bool, err error) {
        matched, err := regexp.MatchString(pattern, s)
        if err != nil {
                panic(err)
        }
        return err
}

Same applies to all such cases below.